src/Orleans/Configuration/GlobalConfiguration.cs

```diff
@@ -415,7 +460,14 @@ internal GlobalConfiguration()
             ExpectedClusterSizeConfigValue = new ConfigValue<int>(DEFAULT_LIVENESS_EXPECTED_CLUSTER_SIZE, true);
             ServiceId = Guid.Empty;
             DeploymentId = Environment.UserName;
-            DataConnectionString = "";
+
+            DataConnectionStringForLiveness = "";
+            // Assume the ado invariant is for sql server storage if not explicitly specified
+            AdoInvariantForLiveness = AdoNetInvariants.InvariantNameSqlServer;
```

Comment:
I guess this is just a matter of how to solve this case. Some ideas:
* If provider invariant is not given in the provider, use the default (SQL Server).
* Hard-code the string "System.Data.SqlClient" to Orleans core to be used if nothing is provided in configuration.
* Leave ``AdoNetInvariants`` to core and move other things outside, or perhaps split it in some fashion.

This is a problem to solve right now, but something to think about when moving ADO.NET system store code outside of core like Azure and ZooKeeper ones. I'm a bit tedious with this, but hopefully makes easier for "outsiders" to follow. :)